src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -140,6 +142,32 @@ PartitionTableNoLock(Oid relationId)
 }
 
 
+/*
+ * try_relation_open_nolock opens a relation with given relationId without
+ * acquiring locks. PostgreSQL's try_relation_open() asserts that caller
+ * has already acquired a lock on the relation, which we don't always do.
+ */
```

Comment:
Maybe add a half a sentence comment saying that this should be in sync with postgres' `try_relation_open` function.. In case anyone has any issues with comes to that?

Reply:
done